src/utility/svd.jl

```diff
@@ -0,0 +1,240 @@
+using TensorKit:
+    SectorDict,
+    _tsvd!,
+    _empty_svdtensors,
+    _compute_svddata!,
+    _create_svdtensors,
+    NoTruncation,
+    TruncationSpace
+CRCExt = Base.get_extension(KrylovKit, :KrylovKitChainRulesCoreExt)
+
+"""
+    PEPSKit.tsvd(t::AbstractTensorMap, alg; trunc=notrunc(), p=2)
+
+Wrapper around `TensorKit.tsvd` which dispatches on the `alg` argument.
+This is needed since a custom adjoint for `PEPSKit.tsvd` may be defined,
+depending on the algorithm. E.g., for `IterSVD` the adjoint for a truncated
+SVD from `KrylovKit.svdsolve` is used.
+"""
+function PEPSKit.tsvd(
+    t::AbstractTensorMap, alg; trunc::TruncationScheme=notrunc(), p::Real=2
+)
+    return TensorKit.tsvd(t; alg, trunc, p)
+end
```

Comment:
This should probably be `tsvd!`. In TensorKit as well, we abuse the fact that `svd!` is not really an in-place function, rather it is a function that destroys its input by using it as workspace. Because of that reason, we assume that if we call `svd!(A)`, afterwards accessing `A` is undefined behaviour, and no guarantees are made on what the contents of `A` will be. Thus, we hack into this by making `svd!` act like `svd` whenever it is being handled by AD.
Effectively, this means that throughout the CTMRG we don't need to use copies, while only the iteration that is being differentiated gets an additional copy.

Reply:
Thanks for the hint, that is an important one. Especially with big bond dimensions this should make difference for the CTMRG performance.